tests/integration/golang/aim/run/update_run_test.go

```diff
@@ -0,0 +1,130 @@
+//go:build integration
+
+package run
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/aim/request"
+	"github.com/G-Research/fasttrackml/pkg/api/aim/response"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type UpdateRunTestSuite struct {
+	suite.Suite
+	client             *helpers.HttpClient
+	runFixtures        *fixtures.RunFixtures
+	experimentFixtures *fixtures.ExperimentFixtures
+	run                *models.Run
+}
+
+func TestUpdateRunTestSuite(t *testing.T) {
+	suite.Run(t, new(UpdateRunTestSuite))
+}
+
+func (s *UpdateRunTestSuite) SetupTest() {
+	s.client = helpers.NewAimApiClient(helpers.GetServiceUri())
+
+	runFixtures, err := fixtures.NewRunFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.runFixtures = runFixtures
+
+	expFixtures, err := fixtures.NewExperimentFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.experimentFixtures = expFixtures
+
+	exp := &models.Experiment{
+		Name:           uuid.New().String(),
+		LifecycleStage: models.LifecycleStageActive,
+	}
+	_, err = s.experimentFixtures.CreateExperiment(context.Background(), exp)
+	assert.Nil(s.T(), err)
+
+	runs, err := s.runFixtures.CreateRuns(context.Background(), exp, 1)
+	assert.Nil(s.T(), err)
+	s.run = runs[0]
+}
+
+func (s *UpdateRunTestSuite) Test_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.runFixtures.UnloadFixtures())
+	}()
+	newName := fmt.Sprintf("%v%v", s.run.Name, "-new")
+	newStatus := string(models.StatusFinished)
+	archived := true
+	tests := []struct {
+		name     string
+		request  request.UpdateRunRequest
+		verifyFn func(*testing.T, response.Success)
+	}{
+		{
+			name: "UpdateOneRun",
+			request: request.UpdateRunRequest{
+				RunID:    &(s.run.ID),
+				Name:     &newName,
+				Status:   &newStatus,
+				Archived: &archived,
+			},
+			verifyFn: func(t *testing.T, resp response.Success) {
```

Comment:
why do we need such function? why we can't verify this moment in test loop?:
```
s.T().Run(tt.name, func(T *testing.T) {
 			var resp response.Success
 			err := s.client.DoPutRequest(
 				fmt.Sprintf("/runs/%s", *tt.request.RunID),
 				tt.request,
 				&resp,
 			)
 			assert.Nil(s.T(), err)
 			if tt.verifyFn != nil {
 				tt.verifyFn(s.T(), resp)
 			}
 		})
```

Reply:
removed -- must have had a reason at some point 🤷